src/dftbp/dftbplus/mainio.F90

```diff
@@ -4014,60 +4023,75 @@ subroutine writeMdOut2(fd, tStress, tPeriodic, tBarostat, isLinResp, eField, tFi
     integer :: ii
     character(lc) :: strTmp
 
-    if (tStress) then
-      if (tBarostat) then
-        write(fd, "(A)") 'Lattice vectors (A)'
+    if (isPeriodic) then
```

Comment:
What happens if helical (the boundary derivatives could be useful there as well)? Might be better to pass in the TboundaryConditions variable in main and test for boundaryCond%iBoundaryCondition .not. boundaryConditions%cluster or only test withBarostat and print lattice vectors then (assuming withBarostat is set for any future helical case).

Reply:
Not entirely sure about the helical case and which quantities are affected. I guess the way the lattice vectors are printed (and converted unit-wise) in case of helical boundary conditions is already broken. Also the cell volume is always zero? What about all the other quantities, do you have a suggestion how to fix it elegantly?

Reply:
We don't change the helical vectors during a calculation at the moment, so can just leave this case out and fix later. 
For reference, it is a (3,1) array, where the first two values are a distance and an angle, while the last value is an integer stored as a real.
The longer term solution is probably to put a writer routine as a method for each boudary condition.

Reply:
I have now changed two things:

1. the total stress is only printed if printing forces is requested, as of your suggestion
2. the block that prints the lattice vectors moved into the `isPeriodic` block, as an extra layer of protection to prevent printing something with wrong unit descriptor (and utilized conversion factor) in case of helical boundary conditions.

Do you agree with those changes or did I overlook something?